extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3TimestampVersionedDataFinder.java

```diff
@@ -67,7 +67,7 @@ public URI call() throws Exception
               final S3Coords coords = new S3Coords(checkURI(uri));
               long mostRecent = Long.MIN_VALUE;
               URI latest = null;
-              S3Object[] objects = s3Client.listObjects(coords.bucket, coords.path, "/");
+              S3Object[] objects = s3Client.listObjects(coords.bucket, coords.path, null);
```

Comment:
oh, nevermind, I misread the javadoc. It says that the delimiter should be ignored for the pattern checking.

But, it also says the /-delimited "parent" of the uri will be checked, which the implementation doesn't seem to be doing (if I am reading things right). It seems to be using the uri as a prefix without adjusting to look at the parent.

IMO the behavior in the implementation makes more sense to me, since why would a user want the documented behavior (modifying their base uri to only look at the parent) instead of using the prefix/directory they provided as-is?